Pipelines in odc

Preeticp opened the pull request:
- This PR primarily adds the Developer console docs for pipelines. Here is the preview for the same: https://odc-pipeline-flows--ocpdocs.netlify.app/openshift-enterprise/latest/pipelines/working-with-pipelines-using-the-developer-perspective.html
- It also renames the 'Creating applications with OpenShift Pipelines' Assembly  to 'Creating CI/CD solutions for applications using OpenShift Pipelines'. 
- It applies to 4.5
- It has been reviewed by the SME, QE, and peer reviewed by the devtools docs team.

andrewballantyne commented on modules/op-constructing-pipelines-using-existing-tasks.adoc:
```diff
@@ -0,0 +1,43 @@
+. Similarly, add an Input resource for the *openshift-client* Task.
+. Click *Create* to create the Pipeline. You are redirected to the *Pipeline Details* page, displaying the details of the created pipeline. You can now use the *Action* button to start the Pipeline.
+
+Optionally, you can also use the *Edit YAML* link, on the upper right of the *Pipeline Builder* page, to directly modify the Pipeline YAML file in the console. You can also use the operator-installed, reusable snippets and samples to craft better Pipelines.
```
```suggestion
Optionally, you can also use the *Edit YAML* link, on the upper right of the *Pipeline Builder* page, to directly modify a Pipeline YAML file in the console. You can also use the operator-installed, reusable snippets and samples to craft better Pipelines.
```
Small but important nit. It's not _the_ Pipeline YAML it's just a template we have... first pass (new creations) is the template and on edit Pipeline it's just back to the YAML that is tied to the Pipeline resource you're editing.

andrewballantyne commented on modules/op-deleting-pipelines.adoc:
```diff
@@ -0,0 +1,8 @@
+
+[id="op-deleting-pipelines_{context}"]
+= Deleting Pipelines
+. In the *Pipelines* view of the *Developer* perspective, click the *Options* {Kebab} menu adjoining a pipeline, and select *Delete Pipeline*.
```
```suggestion
. In the *Pipelines* view of the *Developer* perspective, click the *Actions* {Kebab} menu adjoining a pipeline, and select *Delete Pipeline*.
```
We label the Kebab menu Actions on the details pages, perhaps use that instead of Options?

andrewballantyne replied:
Maybe don't do this - you use `Options` everywhere. I'll leave it around for you to think on it... but I personally think consistency with the rest of the docs makes sense over this different naming.

Preeticp replied:
I will leave this as Options. OCP docs refer to Kebab menu as options menu as a convention, and the curly brackets will render to show the 3 vertical dots. Also Actions menu is on the details page, and not on this one. I have added that point in the last step of the Constructing module.

andrewballantyne commented on modules/op-interacting-with-pipelines-using-the-developer-perspective.adoc:
```diff
@@ -11,36 +11,15 @@ The *Pipelines* view in the *Developer* perspective lists all the Pipelines in a
 . In the *Pipelines* view of the *Developer* perspective, select a project from the *Project* drop-down list to see the Pipelines in that project.
 +
-//image::op-pipeline-list.png[Pipelines in the Developer perspective]
+image::op-pipeline-list.png[Pipelines in the Developer perspective]
```
The admin side changed our filters for the list pages... just FYI, it looks differently.

Preeticp replied:
Updated this and another image and debugged the couple of images that did not render earlier.

Preeticp commented on pipelines/working-with-pipelines-using-the-developer-perspective.adoc:
```diff
@@ -7,19 +7,33 @@ include::modules/pipelines-document-attributes.adoc[]
+The *Pipeline* option enables you to custom create a pipeline as per your needs using the *Pipeline Builder*.
+The xref:../applications/application_life_cycle_management/odc-creating-applications-using-developer-perspective.adoc#odc-importing-codebase-from-git-to-create-application_odc-creating-applications-using-developer-perspective[*From Git*] option enables you to create pipelines using operator-installed pipeline templates and resources while creating an application on {product-title}.
+
+After you create Pipelines, you can view and visually interact with your deployed Pipelines in the *Pipelines* view and the *Topology* view.
```
Thanks @andrewballantyne I have updated the sentences in this section. Moved the Topology sections and the screenshot to the Interacting section, and added a note to make this clear. Hope that helps.

Preeticp commented:
@nkakkar81 @gajanan194 request you to review the PR. Here's the preview for the same: https://odc-pipeline-flows--ocpdocs.netlify.app/openshift-enterprise/latest/pipelines/working-with-pipelines-using-the-developer-perspective.html

gajanan194 commented:
/lgtm

andrewballantyne commented on modules/op-constructing-pipelines-using-existing-tasks.adoc:
```diff
@@ -0,0 +1,43 @@
++
+. A *Missing Resources* warning is displayed on a Task if the resources for the Task are not specified. Click the *s2i-nodejs* Task to see the side panel with details for the Task.
++
+image::op-pipeline-builder-task-details.png[Tasks details in Pipelines Builder]
```
FYI: Your screenshot has the old syntax of param referencing (from before beta resources / tech preview operator)

andrewballantyne commented on pipelines/working-with-pipelines-using-the-developer-perspective.adoc:
```diff
@@ -7,19 +7,33 @@ include::modules/pipelines-document-attributes.adoc[]
 
+include::modules/op-constructing-pipelines-using-existing-tasks.adoc[leveloffset=+1]
+
+include::modules/op-starting-pipelines.adoc[leveloffset=+1]
+
 include::modules/op-interacting-with-pipelines-using-the-developer-perspective.adoc[leveloffset=+1]
```
Maybe flip these two sections? You discuss Pipeline Run details before Pipeline. Not an issue of course, but not the flow my mind would follow.

```suggestion
include::modules/op-interacting-with-pipelines-using-the-developer-perspective.adoc[leveloffset=+1]

include::modules/op-starting-pipelines.adoc[leveloffset=+1]
```

andrewballantyne commented on modules/op-interacting-with-pipelines-using-the-developer-perspective.adoc:
```diff
@@ -11,36 +11,29 @@ The *Pipelines* view in the *Developer* perspective lists all the Pipelines in a
-. In the *Pipeline Runs* section, click *Start Last Run* to start a new pipeline run with the same parameters and resources as the previous ones. This option is disabled if a pipeline run has not been initiated.
+.. On the left navigation panel, click *Topology*, and click on the application to see the Pipeline Runs listed in the side panel.
+.. In the *Pipeline Runs* section, click *Start Last Run* to start a new Pipeline Run with the same parameters and resources as the previous ones. This option is disabled if a Pipeline Run has not been initiated.
 +
```
Wonder if it's worth making a `[NOTE]` here that they can click on the Pipeline link or maybe view the "Starting a Pipeline" section? (this is probably only needed if you follow my previous suggestion on flipping the sections).

Preeticp replied:
So as per the above suggestion I flipped the sections, so now we have Creating via Builder>Creating via 'from git' (just a link)>interacting and then Start.
Therefore I had to move the topology section from the Interacting section to the Start section, as without starting a pipeline you will not see it in the Topology view. Hence note is not required.

nkakkar81 commented on modules/op-constructing-pipelines-using-existing-tasks.adoc:
```diff
@@ -0,0 +1,43 @@
+// *openshift_pipelines/working-with-pipelines-using-the-developer-perspective.adoc
+
+[id="op-constructing-pipelines-using-existing-tasks_{context}"]
+= Constructing Pipelines using existing Tasks and ClusterTasks
```
Does it make sense to rename it to - Constructing Pipelines using Pipeline Builder?

Preeticp replied:
Yeah makes sense, since it is the key point in this module

nkakkar81 commented on modules/op-constructing-pipelines-using-existing-tasks.adoc:
```diff
@@ -0,0 +1,43 @@
+. In the *Add* view of the *Developer* perspective, click the *Pipeline* tile to see the *Pipeline Builder* page.
+. Enter a unique name for the Pipeline.
+. Select a Task from the *Select task* list to add a Task to the Pipeline. For this example, select a *s2i-nodejs* Task.
+.. To add sequential Tasks to the Pipeline, click the + icon to the right or left of the Task, and from the *Select task* list, select the Task you want to add to the Pipeline. For this example, use the + icon to the right of the *s2i-nodejs* Task to add an *openshift-client* Task.
```
Can we add a + icon image, instead of + sign?

Preeticp replied:
Simply replaced using the word plus instead of the symbol to try and use screenshots where it is likely to cause confusion.

Preeticp commented on pipelines/working-with-pipelines-using-the-developer-perspective.adoc:
```diff
@@ -7,19 +7,33 @@ include::modules/pipelines-document-attributes.adoc[]
 toc::[]
 
-You can use the *Developer* perspective of the {product-title} web console to create CI/CD Pipelines for your software delivery process while creating an application on {product-title}.
+You can use the *Developer* perspective of the {product-title} web console to create CI/CD Pipelines for your software delivery process.
```
I will leave this as is. I have changed the next sentences. Changing this too would make it repetitive. Also, since it is an introduction and another option to create pipeline apart from the CLI way. You can seems alright.

Preeticp commented on modules/op-constructing-pipelines-using-existing-tasks.adoc:
```diff
@@ -0,0 +1,43 @@
+* Construct a Pipeline flow using existing Tasks. When you install the OpenShift Pipelines Operator, it adds reusable Pipeline ClusterTasks to your cluster.
+* Specify the type of Resources required for the Pipeline execution, and if required, add additional Parameters to the Pipeline.
+* Reference these Pipeline Resources in each of the Tasks in the Pipeline as Input and Output Resources.
+* The Parameters for a Task are prepopulated based on the specifications of the Task. If required, reference the additional Parameters added to the Pipeline in the Task.
```
I had used the hyphen in the Health Checks flow and I was told to remove it :) Let's follow IBM style guide.

Preeticp commented:
@andrewballantyne I have fixed the nits, and made some structural changes, post brainstorming with @nkakkar81. She is our Pipelines writer and will be doing another round of peer review.

nkakkar81 commented on modules/op-starting-pipelines.adoc:
```diff
@@ -0,0 +1,48 @@
+[id="op-starting-pipelines_{context}"]
+= Starting Pipelines
+
+After you create a Pipeline, you need to start it for the Pipeline to run and execute its Tasks. You can start a Pipeline Run from the *Pipelines* view, *Pipeline Details* page, or the *Topology* view.
```
s/After you create a Pipeline,you need to start it for the Pipeline to run and execute its Tasks./After you create a Pipeline, you need to start it in order to execute the included Tasks in the defined sequence ?

nkakkar81 commented on modules/op-starting-pipelines.adoc:
```diff
@@ -0,0 +1,48 @@
++
+[NOTE]
+====
+For Pipelines created using the *From Git* option, the *Start Pipeline* dialog box displays an `APP_NAME` field in the *Parameters* section, in addition. All the fields in the dialog box are prepopulated by the Pipeline templates in the *From Git* flow.
```
s/the *Start Pipeline* dialog box displays an `APP_NAME` field in the *Parameters* section, in addition./ the *Start Pipeline* dialog box displays an additional `APP_NAME` field in the *Parameters* section.

nkakkar81 commented on modules/op-starting-pipelines.adoc:
```diff
@@ -0,0 +1,48 @@
+.. If required, modify the *Advanced Options* and add credentials to authenticate the specified Git server or Docker registry.
+
+. Click *Start* to start the PipelineRun.
+. The *Pipeline Run Details* page displays the Pipeline being executed. After the Pipeline starts, the Tasks and steps within each Task are executed.
```
s/steps/Steps.
Since Step in a concept, its with capital `S`, see About Tasks topic.  Please change the occurrences of step to Step wherever we refer to Step inside a Task. 

There is one place in the "Understanding Pipelines" section, where step is used, can you change it as part of this PR

Preeticp replied:
Done, it was in the concepts module.

nkakkar81 replied:
Thank you!

nkakkar81 commented on modules/op-editing-pipelines.adoc:
```diff
@@ -0,0 +1,14 @@
+. In the *Pipeline Builder* page:
+* You can add additional Tasks, parameters, or resources to the Pipeline.
+* You can click on the Task you want to modify to see the Task details in the side panel and modify the required Task details, such as the display name, parameters and resources.
+* Alternatively, to delete the Task, click on the Task, and in the side panel, click *Actions* and select *Remove Task*.
```
s/click on/click

Preeticp commented on modules/op-constructing-pipelines-using-pipeline-builder.adoc:
```diff
@@ -0,0 +1,43 @@
++
+. Click *Add Resources* to specify the name and type of resources that the Pipeline will need to use during execution. These resources are then used by the Tasks in the Pipeline as inputs and outputs.
+For this example:
+.. Add an input resource. In the *Name* field, enter `Source`, and from the *Resource Type* drop-down list, select *Git*.
```
As a convention, we mention the location of action first and then the action so that the user knows which section he needs to focus on to complete the action.

Preeticp commented on modules/op-constructing-pipelines-using-pipeline-builder.adoc:
```diff
@@ -0,0 +1,43 @@
+
+.. In the *Input Resources* -> *Source* section, click the *Select Resources* list to see the resources you added to the Pipeline displayed as options, and select *Source*.
+.. In the *Output Resources* -> *Image* section, click the *Select Resources* list, and select *Img*.
+.. If required, in the *Parameters* section, add more parameters to the default ones, by using the `$(params.<param-name>)` syntax.
```
I was told, the UI for this is being improved, so for now these instructions should be good.

Preeticp commented:
@nkakkar81 I have fixed most of the comments and reworded, and have left a couple of comments.
The last commit also changes the title of our main flow in the topic map as well as the Assembly. PTAL and add a LGTM if the changes look good.

nkakkar81 commented:
/lgtm

bmcelvee commented on _topic_map.yml:
```diff
@@ -773,8 +773,8 @@ Topics:
 - Name: Uninstalling OpenShift Pipelines
   File: uninstalling-pipelines
-- Name: Creating applications with OpenShift Pipelines
-  File: creating-applications-with-cicd-pipelines
```
This just came up yesterday - because our redirects list is getting massive, Vikram asked that we avoid changing the files as much as possible and just change the name.

Preeticp replied:
Ack, fixed, and reverted all changes wrt to this change.

bmcelvee commented on modules/op-constructing-pipelines-using-pipeline-builder.adoc:
```diff
@@ -0,0 +1,43 @@
+[id="op-constructing-pipelines-using-pipeline-builder_{context}"]
+= Constructing Pipelines using the Pipeline Builder
+
+You can use the Pipeline Builder to:
```
Maybe add a statement or prerequisite that this procedure requires access to the console.

Preeticp replied:
Modified this statement. Prerequisites with links have been added in the Assembly for this flow.

bmcelvee commented on modules/op-constructing-pipelines-using-pipeline-builder.adoc:
```diff
@@ -0,0 +1,43 @@
+
+You can use the Pipeline Builder to:
+
+* Construct a Pipeline flow using existing Tasks and CLusterTasks. When you install the OpenShift Pipelines Operator, it adds reusable Pipeline ClusterTasks to your cluster.
```
s/CLusterTask/ClusterTask

bmcelvee commented on modules/op-constructing-pipelines-using-pipeline-builder.adoc:
```diff
@@ -0,0 +1,43 @@
+
+. In the *Add* view of the *Developer* perspective, click the *Pipeline* tile to see the *Pipeline Builder* page.
+. Enter a unique name for the Pipeline.
+. Select a Task from the *Select task* list to add a Task to the Pipeline. For this example, select a *s2i-nodejs* Task.
```
I would suggest rewriting this sentence: 
"For this example, select a *s2i-nodejs* Task."

To something like:
"This example uses a *s2i-nodejs* Task." or "This example is based on selecting a *s2i-nodejs* Task."

bmcelvee commented on modules/op-constructing-pipelines-using-pipeline-builder.adoc:
```diff
@@ -0,0 +1,43 @@
+.. To add sequential Tasks to the Pipeline, click the plus icon to the right or left of the Task, and from the *Select task* list, select the Task you want to add to the Pipeline. For this example, use the plus icon to the right of the *s2i-nodejs* Task to add an *openshift-client* Task.
+.. To add a parallel Task to the existing Task, click the plus icon displayed below the Task, and from the *Select Task* list, select the parallel Task you want to add to the Pipeline.
++
+image::op-pipeline-builder.png[Pipeline Builder]
```
Add a figure title

bmcelvee commented on modules/op-constructing-pipelines-using-pipeline-builder.adoc:
```diff
@@ -0,0 +1,43 @@
++
+image::op-pipeline-builder.png[Pipeline Builder]
++
+. Click *Add Resources* to specify the name and type of resources that the Pipeline will use during execution. These resources are then used by the Tasks in the Pipeline as inputs and outputs.
```
General note about the term, "execute/execution" - the IBM style guide advises against using it: 
"executable adj
Use only as an adjective, followed by the noun that it is modifying (for example, “executable program” or
“executable routine”).
execute v
Use only in technical documentation and only if a simpler, less technical term, such as “run” or “issue,” or a
more specific term, such as “start” or “enter,” is inappropriate. If “execute” is used as a keyword, you can
use it in related text to maintain consistency."

Maybe "when running" could work? 

It comes up a couple of times throughout this PR, and I have had this suggestion on previous reviews of my PRs too :)

bmcelvee commented on modules/op-constructing-pipelines-using-pipeline-builder.adoc:
```diff
@@ -0,0 +1,43 @@
++
+. A *Missing Resources* warning is displayed on a Task if the resources for the Task are not specified. Click the *s2i-nodejs* Task to see the side panel with details for the Task.
++
+image::op-pipeline-builder-task-details.png[Tasks details in Pipelines Builder]
```
Add a figure title

bmcelvee commented on modules/op-constructing-pipelines-using-pipeline-builder.adoc:
```diff
@@ -0,0 +1,43 @@
+For this example:
+.. Add an input resource. In the *Name* field, enter `Source`, and from the *Resource Type* drop-down list, select *Git*.
+.. Add an output resource. In the *Name* field, enter `Img`, and from the *Resource Type* drop-down list, select *Image*.
+. The *Parameters* for a Task are prepopulated based on the specifications of the Task. If required, use the *Add Parameters* link to add additional parameters .
```
s/parameters ./parameters.

bmcelvee commented on modules/op-deleting-pipelines.adoc:
```diff
@@ -0,0 +1,8 @@
+//
+// *openshift_pipelines/working-with-pipelines-using-the-developer-perspective.adoc
+
+[id="op-deleting-pipelines_{context}"]
```
Add an introductory statement and the procedure heading

bmcelvee commented on modules/op-editing-pipelines.adoc:
```diff
@@ -0,0 +1,14 @@
+// *openshift_pipelines/working-with-pipelines-using-the-developer-perspective.adoc
+
+[id="op-editing-pipelines_{context}"]
+= Editing Pipelines
```
Add an introductory statement and the procedure heading

bmcelvee commented on modules/op-interacting-with-pipelines-using-the-developer-perspective.adoc:
```diff
@@ -11,36 +11,15 @@ The *Pipelines* view in the *Developer* perspective lists all the Pipelines in a
 . In the *Pipelines* view of the *Developer* perspective, select a project from the *Project* drop-down list to see the Pipelines in that project.
 +
-//image::op-pipeline-list.png[Pipelines in the Developer perspective]
+image::op-pipeline-list.png[Pipelines in the Developer perspective]
```
Add a figure title

bmcelvee commented on modules/op-starting-pipelines.adoc:
```diff
@@ -0,0 +1,48 @@
+* Click on a Task to see logs for each of the Steps in the Task.
+* Click the *Logs* tab to see the logs according to the execution sequence of the Tasks and use the *Download* button to download the logs to a text file.
++
+image::op_pipeline_run.png[Pipeline run]
```
Add figure title

bmcelvee commented on modules/op-starting-pipelines.adoc:
```diff
@@ -0,0 +1,48 @@
+.. On the left navigation panel, click *Topology*, and click on the application to see the Pipeline Runs listed in the side panel.
+.. In the *Pipeline Runs* section, click *Start Last Run* to start a new Pipeline Run with the same parameters and resources as the previous ones. This option is disabled if a Pipeline Run has not been initiated.
++
+image::op_pipeline_topology.png[Pipeline topology]
```
Add figure title

bmcelvee commented:
Just a few suggestions, otherwise LGTM

Preeticp commented:
Hi @bmcelvee I have fixed all your comments and merged all the commits. PTAL and merge the PR if you are good with this.

bmcelvee commented:
Looks good over all! :) Just two minor things and then I'll be able to merge: 
1. Squash commits 
2. Looks like some of the procedure module [introductory statements](https://github.com/redhat-documentation/modular-docs/blob/master/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc) are still missing

Thanks!

Preeticp commented:
Ah, I added the introductory sentences as lead in sentences to the steps, after the Procedure heading. Modified and added to them now. 
But, I wonder if they add any additional value considering the guidelines on Row 16 on Lead-in sentence format, here: https://docs.google.com/spreadsheets/d/1m_kqrFGxTE445qKzT3u8QsfYkTBZjgzKIHQIcXrxjso/edit?ts=5ebd45ae#gid=2022541005

bmcelvee commented:
/cherrypick enterprise-4.5
